test/components/Charts.test.js

```diff
@@ -204,26 +159,29 @@ describe('<Charts />', () => {
         expect(chartNode1.prop("dataSet")).toEqual(testProps.dataSets.get('id1'));
         expect(chartNode1.prop("applicationId")).toBe(testProps.applicationId);
         expect(chartNode1.prop("versionId")).toBe('id1');
-        expect(chartNode1.prop("title")).toBe('v1');
+        expect(chartNode1.prop("title")).toBe('id1');
         expect(chartNode1.prop("onDelete")).toEqual(testProps.onDelete);
+
         expect(chartNode2.type()).toBe(ChartWithHeader);
         expect(chartNode2.prop("viewPortDateRange")).toEqual(testProps.viewPortDateRange);
         expect(chartNode2.prop("dataSet")).toEqual(testProps.dataSets.get('id2'));
-        expect(chartNode2.prop("title")).toBe('v2');
+        expect(chartNode2.prop("title")).toBe('id2');
         expect(chartNode2.prop("versionId")).toBe('id2');
         expect(chartNode2.prop("applicationId")).toBe(testProps.applicationId);
         expect(chartNode2.prop("onDelete")).toEqual(testProps.onDelete);
     });
 
-    it('renders the only matching version data and handles child components', () => {
+    it('renders the matching version with dataSet and does not renders non-matching version and renders provided children', () => {
+        let dataSets = Immutable.Map();
+        dataSets = dataSets.set('id1', createRandomData());
         const testProps = {
-            selectedVersions : [{value: 'id1', text: 'v1'}, {value: 'id2', text: 'v2'}],
+            selectedVersions : [{value: 'id1', text: 'id1'}, {value: 'id2', text: 'id2'}],
             onDelete : () => {},
             viewPortDateRange : {
                 startDate : moment().subtract(1, "days").toDate(),
                 endDate : new Date(Date.now())
             },
-            dataSets : (new Map()).set('id1', createRandomData()),
+            dataSets : dataSets,
```

Comment:
If key name and variable name of the value are the same, you can omit the value.

Ex.:
~~~  javascript
const x = 10;
return {x}; // same as {x: x}
~~~